ego4d/internal/university_files.py

```diff
@@ -0,0 +1,721 @@
+            participant_dict = load_dataclass_dict_from_csv(
+                local_file_path,
+                Particpant,
+                "participant_id",
+            )
+
+        # Load synchronized_videos.csv, if available
+        file_name = "synchronized_videos.csv"
+        synchronized_video_dict = {}
+        if file_name in available_files:
+            if s3_bucket_name:
+                local_file_path = f"{tempdir}/{file_name}"
+                s3_bucket.get_file(f"{metadata_folder_prefix}/{file_name}", local_file_path)
+            else:
+                local_file_path = f"{metadata_folder_prefix}/{file_name}"
+            synchronized_video_dict = load_dataclass_dict_from_csv(
+                local_file_path,
+                SynchronizedVideos,
+                "video_grouping_id",
+            )
+
+            # Load physical_setting.csv, if available
+        file_name = "physical_setting.csv"
+        physical_setting_dict = {}
+        if file_name in available_files:
+            if s3_bucket_name:
+                local_file_path = f"{tempdir}/{file_name}"
+                s3_bucket.get_file(f"{metadata_folder_prefix}/{file_name}", local_file_path)
+            else:
+                local_file_path = f"{metadata_folder_prefix}/{file_name}"
+            physical_setting_dict = load_dataclass_dict_from_csv(
+                local_file_path,
+                PhysicalSetting,
+                "setting_id",
+            )
+
+            # Load annotations.csv, if available
+        file_name = "annotations.csv"
+        annotations_dict = {}
+        if file_name in available_files:
+            if s3_bucket_name:
+                local_file_path = f"{tempdir}/{file_name}"
+                s3_bucket.get_file(f"{metadata_folder_prefix}/{file_name}", local_file_path)
+            else:
+                local_file_path = f"{metadata_folder_prefix}/{file_name}"
+            annotations_dict = load_dataclass_dict_from_csv(
+                local_file_path,
+                Annotations,
+                "university_video_id",
+            )
+
+    return (
+        video_metadata_dict,
+        video_components_dict,
+        auxiliary_video_component_dict,
+        participant_dict,
+        synchronized_video_dict,
+        physical_setting_dict,
+        annotations_dict,
+    )
+
+def validate_university_files(  # noqa :C901
+    video_metadata_dict: Dict[str, VideoMetadata],
+    video_components_dict: Dict[str, List[VideoComponentFile]],
+    auxiliary_video_component_dict: Dict[str, List[AuxiliaryVideoComponentDataFile]],
+    participant_dict: Dict[str, Particpant],
+    synchronized_video_dict: Dict[str, SynchronizedVideos],
+    physical_setting_dict: Dict[str, PhysicalSetting],
+    annotations_dict: Dict[str, Annotations],
+    devices: Dict[str, Device],
+    component_types: Dict[str, ComponentType],
+    scenarios: Dict[str, Scenario],
+    bucket_name: str,
+    s3
+) -> None:
+    error_message = ""
```

Comment:
Let's return a set of warnings and errors, or simply a `List[Tuple[str, str]]`, e.g. `[("error', "this is an error"), ("warning", "this is a warning")]`